compiler/modules/print.llvm/llvm.bal

```diff
@@ -266,20 +266,46 @@ public class FunctionDecl {
     final FunctionType functionType;
     final string functionName;
     string? gcName = ();
-    final EnumAttribute[] attributes = [];
+    final FunctionEnumAttribute[] functionAttributes = [];
+    final ReturnEnumAttribute[] returnAttributes = [];
+    final ParamEnumAttribute[][] paramAttributes = [];
 
     function init(Context context, string functionName, FunctionType functionType) {
         self.functionName = functionName;
         self.functionType = functionType;
+        foreach var _i in 0..< functionType.paramTypes.length() {
+           self.paramAttributes.push([]); 
+        }
     }
 
     function output(Output out) {
         out.push(functionHeader(self));
     }
 
     public function addEnumAttribute(EnumAttribute attribute) {
-        if self.attributes.indexOf(attribute) == () {
-           self.attributes.push(attribute); 
+        if attribute is FunctionEnumAttribute {
+            self.pushAttributeToContainer(self.functionAttributes, attribute);
+        }
+        else {
+            if attribute[0] == "return" {
```

Comment:
You should be able to write this without casting e.g.

```
if attribute is (readonly & ["return", ReturnEnumAttribute]) {
}
```

If that doesn't work, put a JBUG comment

Reply:
Fixed